File: SciChart.Examples/Examples/Examples2D/ModifyAxisBehaviour/LogarithmicAxisChartView/LogarithmicAxisChartView.m
//******************************************************************************
// SCICHART® Copyright SciChart Ltd. 2011-2019. All rights reserved.
//
// Web: http://www.scichart.com
// Support: support@scichart.com
// Sales:   sales@scichart.com
//
// LogarithmicAxisChartView.m is part of the SCICHART® Examples. Permission is hereby granted
// to modify, create derivative works, distribute and publish any part of this source
// code whether for commercial, private or personal use.
//
// The SCICHART® examples are distributed in the hope that they will be useful, but
// without any warranty. It is provided "AS IS" without warranty of any kind, either
// expressed or implied.
//******************************************************************************

#import "LogarithmicAxisChartView.h"
#import "SCDDataManager.h"

@implementation LogarithmicAxisChartView

- (void)initExample {
    self.xAxis = [self generateLogarithmicAxis];
    self.yAxis = [self generateLogarithmicAxis];
    
    SCIXyDataSeries *dataSeries1 = [[SCIXyDataSeries alloc] initWithXType:SCIDataType_Double yType:SCIDataType_Double];
    dataSeries1.seriesName = @"Curve A";
    SCIXyDataSeries *dataSeries2 = [[SCIXyDataSeries alloc] initWithXType:SCIDataType_Double yType:SCIDataType_Double];
    dataSeries2.seriesName = @"Curve B";
    SCIXyDataSeries *dataSeries3 = [[SCIXyDataSeries alloc] initWithXType:SCIDataType_Double yType:SCIDataType_Double];
    dataSeries3.seriesName = @"Curve C";
    
    SCDDoubleSeries *doubleSeries1 = [SCDDataManager getExponentialCurveWithExponent:1.8 count:100];
    SCDDoubleSeries *doubleSeries2 = [SCDDataManager getExponentialCurveWithExponent:2.25 count:100];
    SCDDoubleSeries *doubleSeries3 = [SCDDataManager getExponentialCurveWithExponent:3.59 count:100];
    
    [dataSeries1 appendValuesX:doubleSeries1.xValues y:doubleSeries1.yValues];
    [dataSeries2 appendValuesX:doubleSeries2.xValues y:doubleSeries2.yValues];
    [dataSeries3 appendValuesX:doubleSeries3.xValues y:doubleSeries3.yValues];

    unsigned int line1Color = 0xFFe8c667;
    unsigned int line2Color = 0xFF68bcae;
    unsigned int line3Color = 0xFFc43360;
    
    SCIFastLineRenderableSeries *line1 = [SCIFastLineRenderableSeries new];
    line1.dataSeries = dataSeries1;
    line1.strokeStyle = [[SCISolidPenStyle alloc] initWithColorCode:line1Color thickness:1.5];
    line1.pointMarker = [self getPointMarkerWithSize:5 colorCode:line1Color];

    SCIFastLineRenderableSeries *line2 = [SCIFastLineRenderableSeries new];
    line2.dataSeries = dataSeries2;
    line2.strokeStyle = [[SCISolidPenStyle alloc] initWithColorCode:line2Color thickness:1.5];
    line2.pointMarker = [self getPointMarkerWithSize:5 colorCode:line2Color];
    
    SCIFastLineRenderableSeries *line3 = [SCIFastLineRenderableSeries new];
    line3.dataSeries = dataSeries3;
    line3.strokeStyle = [[SCISolidPenStyle alloc] initWithColorCode:line3Color thickness:1.5];
    line3.pointMarker = [self getPointMarkerWithSize:5 colorCode:line3Color];
    
    [SCIUpdateSuspender usingWithSuspendable:self.surface withBlock:^{
        [self.surface.xAxes add:self.xAxis];
        [self.surface.yAxes add:self.yAxis];
        [self.surface.renderableSeries add:line1];
        [self.surface.renderableSeries add:line2];
        [self.surface.renderableSeries add:line3];
        [self.surface.chartModifiers add:[SCDExampleBaseViewController createDefaultModifiers]];
        
        [SCIAnimations sweepSeries:line1 duration:3.0 andEasingFunction:[SCICubicEase new]];
        [SCIAnimations sweepSeries:line2 duration:3.0 andEasingFunction:[SCICubicEase new]];
        [SCIAnimations sweepSeries:line3 duration:3.0 andEasingFunction:[SCICubicEase new]];
    }];
}

- (SCIEllipsePointMarker *)getPointMarkerWithSize:(int)size colorCode:(uint)colorCode {
    SCIEllipsePointMarker *pointMarker = [SCIEllipsePointMarker new];
    pointMarker.size = CGSizeMake(size, size);
    pointMarker.strokeStyle = SCIPenStyle.TRANSPARENT;
    pointMarker.fillStyle = [[SCISolidBrushStyle alloc] initWithColorCode:colorCode];
    
    return pointMarker;
}

@end

File: SciChart.Examples/Examples/Examples2D/ModifyAxisBehaviour/LogarithmicAxisChartView/LogarithmicAxisChartView.swift
//******************************************************************************
// SCICHART® Copyright SciChart Ltd. 2011-2019. All rights reserved.
//
// Web: http://www.scichart.com
// Support: support@scichart.com
// Sales:   sales@scichart.com
//
// LogarithmicAxisChartView.swift is part of the SCICHART® Examples. Permission is hereby granted
// to modify, create derivative works, distribute and publish any part of this source
// code whether for commercial, private or personal use.
//
// The SCICHART® examples are distributed in the hope that they will be useful, but
// without any warranty. It is provided "AS IS" without warranty of any kind, either
// expressed or implied.
//******************************************************************************

class LogarithmicAxisChartView: SCDLogarithmicAxisChartViewControllerBase {

    override func initExample() {
        xAxis = generateLogarithmicAxis()
        yAxis = generateLogarithmicAxis()

        let dataSeries1 = SCIXyDataSeries(xType: .double, yType: .double)
        dataSeries1.seriesName = "Curve A"
        let dataSeries2 = SCIXyDataSeries(xType: .double, yType: .double)
        dataSeries2.seriesName = "Curve B"
        let dataSeries3 = SCIXyDataSeries(xType: .double, yType: .double)
        dataSeries3.seriesName = "Curve C"

        let SCDDoubleSeries1 = SCDDataManager.getExponentialCurve(withExponent: 1.8, count: 100)
        let SCDDoubleSeries2 = SCDDataManager.getExponentialCurve(withExponent: 2.25, count: 100)
        let SCDDoubleSeries3 = SCDDataManager.getExponentialCurve(withExponent: 3.59, count: 100)
        
        dataSeries1.append(x: SCDDoubleSeries1.xValues, y: SCDDoubleSeries1.yValues)
        dataSeries2.append(x: SCDDoubleSeries2.xValues, y: SCDDoubleSeries2.yValues)
        dataSeries3.append(x: SCDDoubleSeries3.xValues, y: SCDDoubleSeries3.yValues)
        
        let line1Color: UInt32 = 0xFFe8c667
        let line2Color: UInt32 = 0xFF68bcae
        let line3Color: UInt32 = 0xFFc43360
        
        let line1 = SCIFastLineRenderableSeries()
        line1.dataSeries = dataSeries1
        line1.strokeStyle = SCISolidPenStyle(color: line1Color, thickness: 1.5)
        line1.pointMarker = getPointMarker(size: 5, colorCode: line1Color)

        let line2 = SCIFastLineRenderableSeries()
        line2.dataSeries = dataSeries2
        line2.strokeStyle = SCISolidPenStyle(color: line2Color, thickness: 1.5)
        line2.pointMarker = getPointMarker(size: 5, colorCode: line2Color)
        
        let line3 = SCIFastLineRenderableSeries()
        line3.dataSeries = dataSeries3
        line3.strokeStyle = SCISolidPenStyle(color: line3Color, thickness: 1.5)
        line3.pointMarker = getPointMarker(size: 5, colorCode: line3Color)
        
        SCIUpdateSuspender.usingWith(surface) {
            self.surface.xAxes.add(self.xAxis)
            self.surface.yAxes.add(self.yAxis)
            self.surface.renderableSeries.add(line1)
            self.surface.renderableSeries.add(line2)
            self.surface.renderableSeries.add(line3)
            self.surface.chartModifiers.add(SCDExampleBaseViewController.createDefaultModifiers())
            
            SCIAnimations.sweep(line1, duration: 3.0, easingFunction: SCICubicEase())
            SCIAnimations.sweep(line2, duration: 3.0, easingFunction: SCICubicEase())
            SCIAnimations.sweep(line3, duration: 3.0, easingFunction: SCICubicEase())
        }
    }

    fileprivate func getPointMarker(size: Int, colorCode: UInt32) -> SCIEllipsePointMarker {
        let pointMarker = SCIEllipsePointMarker()
        pointMarker.size = CGSize(width: CGFloat(size), height: CGFloat(size))
        pointMarker.strokeStyle = SCIPenStyle.transparent
        pointMarker.fillStyle = SCISolidBrushStyle(color: colorCode)
        
        return pointMarker
    }
}
